src/Block/BlockContextManager.php

```diff
@@ -139,7 +139,7 @@ private function configureSettings(OptionsResolver $optionsResolver, BlockInterf
                 ->addAllowedTypes('extra_cache_keys', 'array')
                 ->addAllowedTypes('attr', 'array')
                 ->addAllowedTypes('ttl', 'int')
-                ->addAllowedTypes('template', ['string', 'bool'])
+                ->addAllowedTypes('template', ['null', 'string'])
```

Comment:
I think this would be BC break for someone having declared the `template` to `true`.

Reply:
You're right. What do you suggest? We can of course still use `(string) $this->getSetting('template');` in the `getTemplate` method.

Reply:
Since the signature was `getTemplate(): string`, I have some trouble to understand how setting the template to false or true could be valid previously.

Reply:
> You're right. What do you suggest? We can of course still use `(string) $this->getSetting('template');` in the `getTemplate` method.

Imho, we don't have to write `(string) $this->getSetting('template');`  in order to add support for boolean since we want to remove it. This was previously not working with boolean.

The only things we could eventually do is
```
// NEXT_MAJOR: Remove bool.
->addAllowedTypes('template', ['null', 'string', 'bool'])
```
But I still wonder if someone can use false/true without any error.

Reply:
Not when using the `Sonata\BlockBundle\Block\Service\AbstractBlockService::renderResponse` since this accepts a `string` only and will also crash when `null` is passed to that function.